### ISSUE TITLE
Brave websites crashing when updating under multi-users

### ISSUE THREAD

**Issue by stevelaskaridis** [brave/brave-browser#32126]
## Description 

In a multi-user setup (OS users I mean), when I am updating Brave on one user and the other user is logged in, i am subsequently getting page crashes on the other user until I restart the browser. This is happening on every update and it affects all pages (even brave://*). This is on OS X and brave-release. But I think this is affecting all Brave versions.

## Steps to Reproduce

   1. Create two user accounts (e.g. Alice and Bob), both sharing the same installation of Brave Browser.
   2. Alice and Bob are simultaneously logged in the system, having their Brave Browser open.
   3. Upon update of the Bob's browser, Alice's websites wills start crashing, until she restarts the browser.

## Actual result:

![image](https://github.com/brave/brave-browser/assets/5221013/324adc8a-91b7-44e4-9ca8-93be1b909b93)

## Expected result:

The websites shouldn't crash, but there should be a message to Alice that she needs to restart her browser so that changes come into effect.

## Reproduces how often: 

Upon every browser update where multiple users are logged into the same system and sharing the same browser installation.

## Desktop Brave version:

Across various versions, but also the most recent release (1.56.20).
Tested on Apple M1 machine, but may be affecting users across architectures and OSs.

## Android Device details:
   - Install type (ARM, x86): n/a
   - Device type (Phone, Tablet, Phablet): n/a
   - Android version: n/a

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Probably
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? I don't think so.
- Does the issue resolve itself when disabling Brave Rewards? I don't think so.
- Is the issue reproducible on the latest version of Chrome? Haven't checked.

**Comment by stevelaskaridis** [comment]
cc @atuchin-m

**Comment by atuchin-m** [comment]
@mherrmann it looks like the updater problem. Do you have any ideas about that?

**Comment by mherrmann** [comment]
@atuchin-m I don't think this is related to the recent updater problem. The recent update problem only started to appear with 1.58. And it only affected Windows.

**Comment by atuchin-m** [comment]
@mherrmann it's definitely an older issue, but I believe it's also related to the update process

**Comment by mherrmann** [comment]
@atuchin-m yes, I agree that it appears to be related to the updater.

@stevelaskaridis do you see any crash- or error-related output when Alice and Bob launch Brave from the command line? I think I'd do `/Applications/Brave Browser.app/.../Brave --enable-logging=stderr --v=0`.
